torch/lib/THC/generic/THCTensorScatterGather.cu

```diff
@@ -111,16 +111,17 @@ void THCTensor_(scatter)(THCState* state, THCTensor *tensor, int dim, THCudaLong
              "Index tensor must have same dimensions as input tensor");
   THArgCheck(THCTensor_(nDimension)(state, src) == THCTensor_(nDimension)(state, tensor), 4,
              "Input tensor must have same dimensions as output tensor");
-  THLongStorage *indexDims = THCudaLongTensor_newSizeOf(state, index);
-  THArgCheck(THCTensor_(isSize)(state, src, indexDims), 3,
-             "Index tensor must have the same size as input tensor.");
-  THLongStorage_free(indexDims);
 
   for (int d = 0; d < THCTensor_(nDimension)(state, tensor); d++) {
+    int64_t indexSizeD = THCudaLongTensor_size(state, index, d);
     if (d != dim) {
-      THArgCheck(THCTensor_(size)(state, tensor, d) == THCTensor_(size)(state, src, d), 4,
-                 "Input tensor must have same size as output tensor apart from the specified dimension");
+      THArgCheck(indexSizeD <= THCTensor_(size)(state, tensor, d), 3,
+                 "Index tensor must not have larger size than output tensor apart from the specified dimension %d, but got index %s output %s",
+                 dim, THCudaLongTensor_sizeDesc(state, index).str, THCTensor_(sizeDesc)(state, tensor).str);
     }
+    THArgCheck(indexSizeD <= THCTensor_(size)(state, src, d), 3,
```

Comment:
Are you sure this is really true? I think it should be fine to e.g. repeat a tensor using scatter

Reply:
I can't seem to find this assertion before, so I'd assume that the kernel is ok with repeating some elements

Reply:
hmm yea Adam makes a good point here, maybe we can remove this assertion. (sorry i didn't think about this case before)

Reply:
actually ignore my comment. Simon pointed out that `index` is actually for the output tensor, so repeating a tensor using scatter doesn't hold (it doesn't work in an intuitive way):
http://pytorch.org/docs/master/tensors.html?highlight=scatter#torch.Tensor.scatter_